cache/cache.go

```diff
@@ -63,7 +57,14 @@ func restoreCache(src string, s storage.Storage, a archive.Archive) error {
 	go func() {
 		defer writer.Close()
 
-		cw <- s.Get(src, writer)
+		err := s.Get(src, writer)
+
+		if err != nil && fallback != "" {
```

Comment:
Also feel like you should check to see if the fallback points to the same file before trying to run the fallback logic.